xarray/backends/api.py

```diff
@@ -79,6 +83,42 @@ def check_name(name):
         check_name(k)
 
 
+def _validate_attrs(dataset):
+    """`attrs` must have a string key and a value which is either: a number
+    a string, an ndarray or a list/tuple of numbers/strings.
+    """
+    def check_attr(name, value):
+        if isinstance(name, basestring):
+            if not name:
+                raise ValueError('Invalid name for attr: string must be length '
+                                 '1 or greater for serialization to netCDF '
+                                 'files')
+        else:
+            raise TypeError('Invalid name for attr: must be a string for '
+                            'serialization to netCDF files')
+
+        if not isinstance(value, basestring) and not isinstance(value, Number) and \
```

Comment:
Prefer parenthesis `()` to explicit backslashes `\`.

You can group isinstance checks, e.g., `isinstance(value, (basestring, Number, ..))`

Finally: `isinstance(numpy_value, Number)` only works on [NumPy 1.9 or later](https://github.com/numpy/numpy/issues/2951#issuecomment-38599552). Currently we still support NumPy 1.7 or newer. So you should check against `np.number` as well.